lib/evss/disability_compensation_form/form8940.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module EVSS
+  module DisabilityCompensationForm
+    class Form8940
```

Comment:
This class is very similar to `Form4142` and `Form0781`.  Should we refactor these into parent class(es) and subclasses?

Something like this:
```
module EVSS
  module DisabilityCompensationForm
    class FormBase
      def initialize(user, form_content, form_id)
        @user = user
        @veteran_data = form_content.dig('form526', 'veteran')
        @final_output = form_content.dig('form526', form_id)
      end

      def translate
        return nil unless @veteran_data && @final_output
        @final_output['vaFileNumber'] = @user.ssn
        @final_output['veteranSocialSecurityNumber'] = @user.ssn
        @final_output['veteranFullName'] = full_name
        @final_output['veteranDateOfBirth'] = @user.birth_date
        @final_output['email'] = @veteran_data['emailAddress']
        @final_output['veteranPhone'] = @veteran_data['primaryPhone']
        yield
        @final_output.to_json
      end

      private

      def full_name
        {
          "first": @user.first_name,
          "middle": @user.middle_name,
          "last": @user.last_name
        }
      end

     def address(data)
        {
          "city": data['city'],
          "country": data['country'],
          "postalCode": data['zipCode'],
          "street": data['addressLine1'],
          "street2": data['addressLine2'],
          "state": data['state']
        }
      end
    end
  end
end
```
And this file would look like this:
```
module EVSS
  module DisabilityCompensationForm
    class Form8940 < FormBase
      def translate
        super do
          @final_output['veteranAddress'] = address(@veteran_data['mailingAddress'])
        end
      end
    end
  end
end
```

And refactor the other two form classes the same way.

Thoughts @lihanli ?

Reply:
i agree, duplicate code should be refactored

Reply:
I understand that the code is technically duplicated, but shouldn't these be 1:1 with the individual form schemas? Also, there isn't any guarantee that all source form schemas will share these fields or that the target schema won't have valid data that would be lost because the base class was overwriting it due to a shared name.

These are all using the 526ez as the source, so maybe the base class could be 526 specific?

Also, I don't think these aren't named well because they're really form data translators but `Form0781` sounds like it's a an internal model.

Let me know what you'd like me to do! This is somewhat ancillary to the main goal of this PR since the wiring of this form will be handled in another PR.

Reply:
> there isn't any guarantee that ... the target schema won't have valid data that would be lost because the base class was overwriting it due to a shared name.

I don't think I understand.  The subclass overwrites the parent class; not the other way around. 
 Can you give an example?  

> I don't think these aren't named well because they're really form data translators but Form0781 sounds like it's a an internal model.

Did you mean to say 
> I think these aren't named well 

?
I'm all for files having the most accurate names.  Feel free to rename them all.

> shouldn't these be 1:1 with the individual form schemas?

They definitely should produce output congruent with the schema.  This refactor should not affect that.

> Also, there isn't any guarantee that all source form schemas will share these fields

True, but I still think we should refactor with a parent class.  If we add a bunch more forms that look totally different, they don't necessarily have to share the parent class, but these look so similar that I think they should.

Reply:
> > there isn't any guarantee that ... the target schema won't have valid data that would be lost because the base class was overwriting it due to a shared name.
> 
> I don't think I understand. The subclass overwrites the parent class; not the other way around.
> Can you give an example?

Sorry, I'm referring to the data, not the class.

We're converting the data between two different models. Moving those mappings to a base class only works if there is a single base schema (which is only true in this case because all the forms we are currently dealing with are populated from the 526ez) and a common target schema. There isn't any guarantee that the target form schema will have any similarities, so you could potentially have data on the target model that is overwritten by the data on the source model because the mapper inheriting from the base class doesn't have full control over the mappings.

> 
> > I don't think these aren't named well because they're really form data translators but Form0781 sounds like it's a an internal model.
> 
> Did you mean to say
> 
> > I think these aren't named well
> 
> ?
> I'm all for files having the most accurate names. Feel free to rename them all.

Sorry, correct, the name doesn't seem very accurate.  We're building a model that isn't really defined anywhere, but this class isn't the model.

> > shouldn't these be 1:1 with the individual form schemas?
> 
> They definitely should produce output congruent with the schema. This refactor should not affect that.
> 
> > Also, there isn't any guarantee that all source form schemas will share these fields
> 
> True, but I still think we should refactor with a parent class. If we add a bunch more forms that look totally different, they don't necessarily have to share the parent class, but these look so similar that I think they should.

I understand, my point was that just because fields might be common across two different models, the models themselves aren't necessarily related or a good fit for inheritance.  If we name it something like `Base526FormDataPopulator` then it's more clear (at least on the source side).

Reply:
> I understand, my point was that just because fields might be common across two different models, the models themselves aren't necessarily related or a good fit for inheritance. If we name it something like Base526FormDataPopulator then it's more clear (at least on the source side).

I am not familiar with the domain of these forms, so maybe I'm missing something; I am just reviewing from a syntax perspective.  But based on what you've said, they are all used for the same task (transforming data) in the same namespace with the same methods that are all based on the same root form (thereby sharing common input data/schema), right?  
That sounds like a good case for inheritance to me.

I get your point that this parent class would only be good for these `526` based forms.  That's fine and inline with my opinion ('If we add a bunch more forms that look totally different, they don't necessarily have to share the parent class')

Given that, I'm totally fine with naming the parent class `Base526FormDataPopulator` and renaming the subclasses to be more accurate/descriptive. In fact, it would help make things more clear for someone working with this part of the codebase for the first time (such as me 😃).

I appreciate you talking this through with me.  

FYI, I'm signing off now, and will be back on Tuesday morning.

Reply:
Because this isn't actually required for this story, I'm pulling it from the PR and we'll handle this in a future story.